### ISSUE TITLE
Abbreviation-like search

### ISSUE THREAD

**Issue by sergei-dyshel** [nixprime/cpsm#8]
Sometimes, in large repos, it may be helpful to limit pattern to match only beginnings of words. For example, `ctpy` will match `ctrl.py` but won't match `control.py'. How hard is to add such feature to `cpsm`? Which code is responsible for letter-by-letter matching process?

**Comment by nixprime** [comment]
cpsm's ranking algorithm is a bit complicated, but it *already* includes matching at the beginning of words as a very strong signal. Do you have a specific example of bad behavior?

**Comment by sergei-dyshel** [comment]
I agree that cpsm definitely prefers matches at beginning of words but for some case the "signal" is not strong enough. 

Besides, sometimes there is a strange delay of 1-2 which is not tolerable for me.

Anyway, I'm currently working on CtrlP's matcher which will match *only* at beginnings of words. I can send you link when it's published, if you want.
